### ISSUE TITLE
Downloading by sending data through URL doesn't work for larger files

### ISSUE THREAD

**Issue by radumas** [CityofToronto/bdit_data_tools#15]
Because of the way we're hacking passing the data to the Flask `send_file()` function, through data URIs, there's unfortunately an approximately 2MB limit on the size of the URI in Chrome

https://stackoverflow.com/questions/695151/data-protocol-url-size-limitations/41755526#41755526

[Discussion of workarounds on the plot.ly forum](https://community.plot.ly/t/download-raw-data/4700/20?u=rad)

Would it be possible to store the data in a hidden div or a [Store component](https://dash.plot.ly/dash-core-components/store) and then [attach javascript to pull data from that div for download](https://stackoverflow.com/questions/29702758/html-button-to-save-div-content-using-javascript)?

**Comment by radumas** [comment]
Though from [these](https://stackoverflow.com/a/4185028/4047679) [answers](https://stackoverflow.com/a/7439083/4047679) seems that may use the same process.

**Comment by radumas** [comment]
I think just say don't use Chrome for large files and be done with it. Test large files with IE and Firefox

**Comment by chelsrosic** [comment]
After checking the log of the app, I found that the following error was occurring (below is the info from the log): 
```
[2019-02-22 15:41:06,904] ERROR in app: Exception on /_dash-update-component [POST]
Traceback (most recent call last):
  File "/home/crosic/.local/lib/python2.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/crosic/.local/lib/python2.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/crosic/.local/lib/python2.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/crosic/.local/lib/python2.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/crosic/.local/lib/python2.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/crosic/.local/lib/python2.7/site-packages/dash/dash.py", line 967, in dispatch
    return self.callback_map[target_id]['callback'](*args)
  File "/home/crosic/.local/lib/python2.7/site-packages/dash/dash.py", line 922, in add_context
    self._validate_callback_output(output_value, output)
  File "/home/crosic/.local/lib/python2.7/site-packages/dash/dash.py", line 866, in _validate_callback_output
    _validate_value(val, index=i)
  File "/home/crosic/.local/lib/python2.7/site-packages/dash/dash.py", line 849, in _validate_value
    index=index
  File "/home/crosic/.local/lib/python2.7/site-packages/dash/dash.py", line 803, in _raise_invalid
    bad_val=bad_val).replace('    ', ''))
InvalidCallbackReturnValue:
The callback for property `children` of component `output-data-upload`
returned a tree with one value having type `IndexError`
which is not JSON serializable.

The value in question is located at
[0] Div
IndexError

and has string representation
`list index out of range`

In general, Dash properties can only be
dash components, strings, dictionaries, numbers, None,
or lists of those.
```
While this issue occurred, there was no error message being outputted on the screen. 

This error occurred when inputting a CSV file with about 200 rows, and a Excel file with about 300 rows.

**Comment by chelsrosic** [comment]
Also, when I input files that are very large (i.e. a CSV with 6000 records, some of which have rows with entries that are blank), I get the following error: 
`'utf8' codec can't decode bytes in position 15-16: invalid continuation byte`

I tried changing the encoding type of the file in the `read_csv()` function call in the `parse_contents` function of `app.py` to [ISO-8859-1](https://stackoverflow.com/questions/19699367/unicodedecodeerror-utf-8-codec-cant-decode-byte) and [latin-1](https://stackoverflow.com/questions/5552555/unicodedecodeerror-invalid-continuation-byte) and after changing to either the following error occurred: 
`Error tokenizing data. C error: Expected 2 fields in line 3, saw 3`

After doing more [research](https://stackoverflow.com/questions/18039057/python-pandas-error-tokenizing-data), I added the `error_bad_lines=False` parameter to the `read_csv` function. 
But this change resulted in yet another error: 
`Error tokenizing data. C error: Buffer overflow caught - possible malformed input file. `
This error leads me to believe that the encoding could still be wrong, but I am really not sure since I do not understand a lot about encoding and Wikipedia isn't really helpful.
